### PR TITLE
Add licenseRefLookup

### PR DESCRIPTION
**Pull request by qtomlinson**

### CODE REVIEW

**Comment by qtomlinson** [comment]
@elrayle Appreciate your feedback!

File: test/index.js [hunk 1]
```diff
@@ -156,3 +157,169 @@ it('should parse BSD-3-Clause-Modification', function () {
...
+describe('licenseRefLookup', function () {
+  const licenseRefLookup = function (identifier) {
+    if (identifier === 'afpl-9.0') return 'LicenseRef-scancode-afpl-9.0'
+    if (identifier === 'activestate-community') return 'LicenseRef-scancode-activestate-community'
+    if (identifier === 'ac3filter') return 'LicenseRef-scancode-ac3filter'
+  }
+
+  it('should parse single licenseRef', function () {
+    assert.deepEqual(
+      p('afpl-9.0', { licenseRefLookup}),
```
Comment (ljones140) [inline test/index.js]: nit: inconsistent whitespace around {}

File: test/index.js [hunk 2]
```diff
@@ -156,3 +157,169 @@ it('should parse BSD-3-Clause-Modification', function () {
...
+  it('should parse licenseRef within bracket', function () {
+    assert.deepEqual(
+      p('(afpl-9.0)', { licenseRefLookup }),
+      {license: 'LicenseRef-scancode-afpl-9.0'}
+    )
+  })
+
+  it('should parse (licenseRef OR license)', function () {
+    assert.deepEqual(
+      p('(afpl-9.0 OR Apache-2.0)', { licenseRefLookup}),
```
Comment (ljones140) [inline test/index.js]: nit: inconsistent whitespace around {}

File: test/index.js [hunk 3]
```diff
@@ -156,3 +157,169 @@ it('should parse BSD-3-Clause-Modification', function () {
...
+        left: {license: 'Apache-2.0'},
+        conjunction: 'and',
+        right: {license: 'LicenseRef-scancode-afpl-9.0'}
+      }
+    )
+  })
+
+  it('should parse license AND (licenseRef AND licenseRef)', function () {
+    assert.deepEqual(
+      p('Apache-2.0 AND (afpl-9.0 AND afpl-9.0)', { licenseRefLookup}),
```
Comment (ljones140) [inline test/index.js]: nit: inconsistent whitespace around {}

File: test/index.js [hunk 4]
```diff
@@ -156,3 +157,169 @@ it('should parse BSD-3-Clause-Modification', function () {
...
+    assert.deepEqual(
+      p('LicenseRef-scancode-afpl-9.0 AND MIT', { licenseRefLookup }),
+      {
+        left: {license: 'LicenseRef-scancode-afpl-9.0'},
+        conjunction: 'and',
+        right: {license: 'MIT'}
+      }
+    )
+  })
+})
```
Comment (ljones140) [inline test/index.js:324]: Tests look great 👍

**Review by ljones140 — APPROVED** [review]
Looks great.

You may want to sort the whitespace issues before merging

File: test/index.js (hunk 2 above)
Comment (qtomlinson) [reply test/index.js]: updated.

File: test/index.js (hunk 1 above)
Comment (qtomlinson) [reply test/index.js]: updated.

File: test/index.js (hunk 3 above)
Comment (qtomlinson) [reply test/index.js]: updated

**Comment by qtomlinson** [comment]
@ljones140 @elrayle Thanks for the review! spaces have been fixed.  I don't have rights to merge.  Could you please help?

**Review by elrayle — APPROVED** [review]
LGTM; Glad to see this transformation happening here.

**Pull request merged**